tests/js/realm-tests.js

```diff
@@ -1538,8 +1538,9 @@ module.exports = {
     testManualCompactMultipleInstances: function() {
         const realm1 = new Realm({schema: [schemas.StringOnly]});
         const realm2 = new Realm({schema: [schemas.StringOnly]});
+        // realm1 and realm2 are the same Realm instance
         realm2.objects('StringOnlyObject');
-        TestCase.assertFalse(realm1.compact());
+        TestCase.assertTrue(realm1.compact());
```

Comment:
I'm guessing the test here is in the "limitation" of `compact()`: _"Compaction will not occur if other `Realm` instances exist."_ Maybe add a comment about this?

And a question: What happens to `realm2` when `realm1` is closed (e.g. `realm1.close()`)?